src/canmatrix/canmatrix.py

```diff
@@ -425,62 +436,75 @@ def pgn(self, value):
 
     @property
     def priority(self):
+        """Get J1939 priority."""
         return self._j1939_prio
 
     @priority.setter
     def priority(self, value):
+        """Set J1939 priority."""
         self._j1939_prio = value
         self.recalcJ1939Id()
 
     @property
     def source(self):
+        """Get J1939 source."""
         return self._j1939_source
 
     @source.setter
-    def pgn(self, value):
+    def source(self, value):
+        """Set J1939 source."""
         self._j1939_source = value
         self.recalcJ1939Id()
 
     def recalcJ1939Id(self):
+        """Recompute J1939 ID"""
         self.id = (self.j1939_source & 0xff) + ((self.j1939_pgn & 0xffff) << 8) + ((self.j1939_prio & 0x7) << 26)
         self.extended = True
         self.is_j1939 = True
 
     #    @property
-#    def cycleTime(self):
-#        if self._cycleTime is None:
-#            self._cycleTime = self.attribute(None, "GenMsgCycleTime")
-#        return self._cycleTime
-
-#    @property
-#    def sendType(self, db = None):
-#        if self._sendType is None:
-#            self._sendType = self.attribute(None, "GenMsgSendType")
- #       return self._sendType
-
-#    @cycleTime.setter
-#    def cycleTime(self, value):
-#        self._cycleTime = value
-#        self.attributes["GenMsgCycleTime"] = value
-
-#    @sendType.setter
-#    def sendType(self, value):
- #       self._sendType = value
- #       self.attributes["GenMsgCycleTime"] = value
+    #    def cycleTime(self):
+    #        if self._cycleTime is None:
+    #            self._cycleTime = self.attribute(None, "GenMsgCycleTime")
+    #        return self._cycleTime
 
-
-
-    def attribute(self, db, attributeName):
-        if attributeName in self.attributes:
-            return self.attributes[attributeName]
+    #    @property
+    #    def sendType(self, db = None):
+    #        if self._sendType is None:
+    #            self._sendType = self.attribute(None, "GenMsgSendType")
+    #       return self._sendType
+
+    #    @cycleTime.setter
+    #    def cycleTime(self, value):
+    #        self._cycleTime = value
+    #        self.attributes["GenMsgCycleTime"] = value
+
+    #    @sendType.setter
+    #    def sendType(self, value):
+    #       self._sendType = value
+    #       self.attributes["GenMsgCycleTime"] = value
+
+    def attribute(self, name, db=None, default=None):
```

Comment:
This is definitely a breaking change though it does seem better.  I think it's better for `canmatrix` to be tending towards improving so I'm comfortable with this but figured I'd highlight it.

Reply:
If this change is accepted, I'd adopt the similar functions of another classes accordingly.

Reply:
canmatrix is at 0. - version. I agree that improving is always better than staying compatible.  
I'd accept it!

Reply:
I'll refactor the `attribute()` of another classes to have the same signature in another PR (when I don't fall asleep in the train next time)